python/dask_cudf/dask_cudf/io/orc.py

```diff
@@ -23,16 +25,27 @@ def read_orc(path, **kwargs):
     cudf.read_orc
     """
 
-    filenames = sorted(glob(str(path)))
     name = "read-orc-" + tokenize(path, **kwargs)
-
-    meta = cudf.read_orc(filenames[0], **kwargs)
-
-    graph = {
-        (name, i): (apply, cudf.read_orc, [fn], kwargs)
-        for i, fn in enumerate(filenames)
-    }
-
-    divisions = [None] * (len(filenames) + 1)
-
-    return dd.core.new_dd_object(graph, name, meta, divisions)
+    dsk = {}
+    if "://" in str(path):
+        files = open_files(path)
+        with files[0] as fn:
+            meta = cudf.read_orc(fn, **kwargs)
+        parts = [delayed(_read_orc)(f, **kwargs) for f in files]
+        return dd.from_delayed(parts, meta=meta)
```

Comment:
There is some slight complexity added when using dask delayed.  For internal functions we tend to use raw task graphs when possible.  

Maybe something like the following?

```python
dsk = {(name, i): (apply, _read_orc, [f], kwargs) for i, f in enumerate(files)}
divisions = [None] * (len(dsk) + 1)
return dd.DataFrame(dsk, name, meta, divisions)
```

Reply:
ah, right.  I wanted to do this but lost sight of it when I was messing about with contexts and OpenFile